src/helpers/analytics.js

```diff
@@ -0,0 +1,10 @@
+export default function dispatchAnalyticsEvent({type, action, ...data}) {
+  const event = new CustomEvent("component.library.analytics", {
```

Comment:
Can we call this custom event `component-library-analytics`? Seems a bit more "JavaScripty." :man_shrugging: